### ISSUE TITLE
no-unused-vars: wrong error with for of loop

### ISSUE THREAD

**Issue by Mottoweb** [eslint/eslint#12224]
**Tell us about your environment**

* **ESLint Version: 6.3.0**
* **Node Version: 8.16.0**
* **npm Version: 6.4.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "react-app",
  "settings": {
    "import/resolver": {
      "node": {
        "paths": ["src"]
      }
    }
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export function* handlePendingOperationsCompleted(action) {
  const { operations } = action.payload

  const successfulRestorations = getOperationsOfType(operations, operationTypes.RESTORE_ACCOUNT)

  for (let operation of successfulRestorations) {
    yield put(restorationSuccess(operation.accountNumber))
  }
}
```

```bash
./node_modules/eslint/bin/eslint.js src
```

**What did you expect to happen?**
No error for no-unused-vars

**What actually happened? Please include the actual, raw output from ESLint.**
I received an error for no-unused-vars
`operation is defined but never used`
**Are you willing to submit a pull request to fix this bug?**
I might

**Comment by mysticatea** [comment]
Thank you for your report.

This looks the same issue as #12117. Use the latest `babel-eslint`.

I'm closing this issue as a duplicate of #12117.
